### PR TITLE
ICU-22621 Apply straightforward Clang-Tidy fixes

### PR DESCRIPTION
**Pull request by roubert**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22621
- [x] Required: The PR title must be prefixed with a JIRA Issue number.
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number.
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/test/intltest/transtst.cpp [hunk 1]
```diff
@@ -1730,11 +1730,11 @@ void TransliteratorTest::TestRemove() {
     UParseError parseError;
     UErrorCode status = U_ZERO_ERROR;
     Transliterator *t = Transliterator::createInstance("Remove[abc]", UTRANS_FORWARD, parseError, status);
-    if (t == 0) {
+    if (t == nullptr) {
         errln("FAIL: createInstance failed");
         return;
     }
-    
+
```
Comment (eggrobin) [inline icu4c/source/test/intltest/transtst.cpp:1737]: Spurious formatting changes.

**Review by eggrobin — APPROVED** [review]

File: icu4c/source/common/locdispnames.cpp [hunk 2]
```diff
@@ -59,7 +59,7 @@ Locale::getDisplayLanguage(const Locale &displayLocale,
     int32_t length;
 
     buffer=result.getBuffer(ULOC_FULLNAME_CAPACITY);
-    if(buffer==0) {
+    if (buffer == nullptr) {
```
[the remark this answers is not in the document]
Comment (roubert) [reply icu4c/source/common/locdispnames.cpp:62]: > I suppose we don’t care when it is on a line that is touched anyway?

I think so, improved formatting in an expression that's being changed is a bonus.

File: icu4c/source/common/ubrk.cpp [hunk 3]
```diff
@@ -70,11 +70,11 @@ ubrk_open(UBreakIteratorType type,
...
   // check for allocation error
   if (U_FAILURE(*status)) {
-     return 0;
+      return nullptr;
   }
-  if(result == 0) {
-    *status = U_MEMORY_ALLOCATION_ERROR;
-    return 0;
+  if (result == nullptr) {
+      *status = U_MEMORY_ALLOCATION_ERROR;
```
[the remark this answers is not in the document]
Comment (roubert) [reply icu4c/source/common/ubrk.cpp:76]: Done.

File: icu4c/source/test/intltest/transtst.cpp (hunk 1 above)
Comment (roubert) [reply icu4c/source/test/intltest/transtst.cpp:1737]: Done.

File: icu4c/source/test/intltest/ucaconf.cpp [hunk 4]
```diff
@@ -127,20 +127,20 @@ void UCAConformanceTest::openTestFile(const char *type)
...
             uprv_strcpy(buffer+bufLen, "_STUB");
             uprv_strcat(buffer, ext);
             testFile = fopen(buffer, "rb");
 
-            if (testFile == 0) {
+            if (testFile == nullptr) {
                 *(buffer+bufLen) = 0;
                 dataerrln("Could not open any of the conformance test files, tried opening base %s\n", buffer);
-                return;        
+                return;
```
[the remark this answers is not in the document]
Comment (roubert) [reply icu4c/source/test/intltest/ucaconf.cpp:143]: Done.

**Review by eggrobin — APPROVED** [review]

**Pull request merged**